src/openrct2/paint/tile_element/Paint.Surface.cpp

```diff
@@ -1296,6 +1296,10 @@ void surface_paint(paint_session* session, uint8_t direction, uint16_t height, c
         const uint16_t localHeight = height + 16;
         const uint16_t waterHeight = tileElement->AsSurface()->GetWaterHeight();
 
+        if ((session->ViewFlags & VIEWPORT_FLAG_CLIP_VIEW)
+            && (tileElement->AsSurface()->GetWaterHeight() > gClipHeight * COORDS_Z_STEP))
+            return;
```

Comment:
Don't just break out of this function, later on there is code that would draw the park fences. This would hide those. Only skip this section of the paint function when the water gets clipped.

Also, you can use waterHeight here instead of calling the function again. It would be even nicer if you moved that variable just before the parent if-condition and use it there as well.

Reply:
Oh that's true. My bad. Could you check the fix commit?

Reply:
Actually, we can just merge all 3 conditions into one parent if statement as ParkFences check does